src/test/java/it/pagopa/swclient/mil/TerminalResourceTest.java

```diff
@@ -0,0 +1,126 @@
+package it.pagopa.swclient.mil;
+
+import com.mongodb.MongoWriteException;
+import com.mongodb.ServerAddress;
+import com.mongodb.WriteError;
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import io.smallrye.mutiny.Uni;
+import it.pagopa.swclient.mil.controller.model.CommonHeader;
+import it.pagopa.swclient.mil.controller.model.TerminalDto;
+import it.pagopa.swclient.mil.dao.TerminalEntity;
+import it.pagopa.swclient.mil.dao.TerminalRepository;
+import it.pagopa.swclient.mil.util.TerminalTestData;
+import jakarta.ws.rs.WebApplicationException;
+import org.bson.BsonDocument;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static io.restassured.RestAssured.given;
+
+@QuarkusTest
+class TerminalResourceTest {
+
+    @InjectMock
+    TerminalRepository terminalRepository;
+
+    static CommonHeader commonHeader;
+
+    static TerminalDto terminalDto;
+
+    static TerminalEntity terminalEntity;
+
+    @BeforeAll
+    static void createTestObjects() {
+        commonHeader = TerminalTestData.getCorrectCommonHeader();
+        terminalDto = TerminalTestData.getCorrectTerminalDto();
+        terminalEntity = TerminalTestData.getCorrectTerminalEntity();
+    }
+
+    @Test
+    void testCreateTerminalEndpoint() {
+        Mockito.when(terminalRepository.persist(Mockito.any(TerminalEntity.class))).thenReturn(Uni.createFrom().item(terminalEntity));
```

Comment:
here i would expect you mock the service, not the controller. It's always better to test each layer in isolation

Reply:
Resolved in [https://github.com/pagopa/mil-terminal-registry/pull/25/commits/8c55df2932810878f744b7992b7a535c9c819587](https://github.com/pagopa/mil-terminal-registry/pull/url)